### PR TITLE
SDL Localization

### PR DESCRIPTION
**Pull request by kshala-ford**
This proposal is about providing a localization API to developers and make it easy to localize the app based on the language settings of the head unit.

### CODE REVIEW

**Comment by kshala-ford** [comment]
@theresalech PR is ready for review

File: proposals/NNNN-localization.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+this.getResources().getText(R.string.some_key); // loading string from phone language
+```
+
+would be
+
+```java
+getLocalization().getText(R.string.some_key); // loading string from SDL language
+```
+
+The Android library don't contain SDL managers yet. The app developer can use the class manually if needed. Once Android includes a lifecycle manager it can automatically provide a localization object.
```
Comment (theresalech) [inline proposals/NNNN-localization.md]: Please update "don't" to "doesn't".

File: proposals/NNNN-localization.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+### Detailed design
+
+First of all the localization code already exist:
```
Comment (theresalech) [inline proposals/NNNN-localization.md]: Please update to "exists:"

File: proposals/NNNN-localization.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+Resources resources = context.getResources();
+Configuration config = new Configuration(resources.getConfiguration());
+config.setLocale(locale);
+ContextWrapper wrapper = new ContextWrapper(context);
+this.context = wrapper.createConfigurationContext(config);
+```
+
+This proposal is using an API called [createConfigurationContext()](https://developer.android.com/reference/android/content/ContextWrapper.html#createConfigurationContext(android.content.res.Configuration)) which was added at API level 17 (Android 4.2 Jelly Beans MR1). This method creates a new context separated from the apps configuration which allows the app to access resources of another language.
+
+The downside is the API level (see [Android dashboards](https://developer.android.com/about/dashboards/index.html)). As it is expected this number to become lower overtime and as it already reached a low number it is proposed to just use the phone language instead.
```
Comment (theresalech) [inline proposals/NNNN-localization.md]: Can you please change the last sentence to the following?

"As it is expected that this number will become lower over time, and as it has already reached a low number, it is proposed to just use the phone language instead."

File: proposals/NNNN-localization.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+## Impact on existing code
+
+This change whould not affect any existing code as it's adding classes and a single property to the lifecycle manager.
+
+## Alternatives considered
+
+**Change language configuration of the entire app** would be an easy approach (very few lines of code) but causes the UI of the app on the phone to change to the SDL language. This was dropped as it is too agressive.
+
+**Create custom localization files** separate from the existing and known ones. This would make localization is independent of the Android or iOS API  but it's much more effort to follow CLDR specifications especially when it comes to plural rules (which are very helpful for natural voice output). Furthermore the app developer has to do translations outside of the known environment which may not be accepted.
```
Comment (theresalech) [inline proposals/NNNN-localization.md]: Please change the beginning of the first sentence to "This would make localization independent..."

**Comment by kshala-ford** [comment]
Thanks @theresalech. The changes are applied.

**Pull request merged**